app/components/search_bar/search_bar.android.js

```diff
@@ -0,0 +1,174 @@
+    static propTypes = {
+        height: PropTypes.number.isRequired,
+        fontSize: PropTypes.number.isRequired,
+        text: PropTypes.string,
+        placeholder: PropTypes.string,
+        showCancelButton: PropTypes.bool,
+        textFieldBackgroundColor: PropTypes.string,
+        placeholderTextColor: PropTypes.string,
+        textColor: PropTypes.string,
+        onChange: PropTypes.func,
+        onChangeText: PropTypes.func,
+        onFocus: PropTypes.func,
+        onBlur: PropTypes.func,
+        onSearchButtonPress: PropTypes.func,
+        onCancelButtonPress: PropTypes.func
+    };
+
+    static defaultProps = {
+        placeholder: 'Search',
+        showCancelButton: true,
+        placeholderTextColor: '#bdbdbd',
+        textColor: '#212121',
+        onSearchButtonPress: () => true,
+        onCancelButtonPress: () => true,
+        onChangeText: () => true,
+        onFocus: () => true,
+        onBlur: () => true
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isOnFocus: false,
+            value: this.props.text
+        };
+        this.onFocus = this.onFocus.bind(this);
+        this.onBlur = this.onBlur.bind(this);
+        this.onCancelButtonPress = this.onCancelButtonPress.bind(this);
+        this.onSearchButtonPress = this.onSearchButtonPress.bind(this);
+        this.onChangeText = this.onChangeText.bind(this);
+    }
+
+    onSearchButtonPress() {
+        const onSearchButtonPress = this.props.onSearchButtonPress;
+        if (this.state.value) {
+            onSearchButtonPress(this.state.value);
+        }
+    }
+
+    onCancelButtonPress() {
+        const onCancelButtonPress = this.props.onCancelButtonPress;
+        this.setState({
+            isOnFocus: false,
+            value: ''
+        });
+
+        onCancelButtonPress();
+        Keyboard.dismiss();
+    }
+
+    onChangeText(value) {
+        const onChangeText = this.props.onChangeText;
+        this.setState({value});
+        onChangeText(value);
+    }
+
+    onFocus() {
+        const onFocus = this.props.onFocus;
+        this.setState({isOnFocus: true});
+        onFocus();
+    }
+
+    onBlur() {
+        const onBlur = this.props.onBlur;
+        this.setState({isOnFocus: false});
+        onBlur();
+        Keyboard.dismiss();
+    }
+
+    render() {
+        const {
+            height,
+            placeholder,
+            fontSize,
+            placeholderTextColor,
+            textColor,
+            textFieldBackgroundColor
+        } = this.props;
+
+        const searchBarStyle = {
+            height: height + 10,
+            paddingLeft: height * 0.25,
+            backgroundColor: textFieldBackgroundColor
+        };
+
+        const inputStyle = {
+            paddingLeft: height * 0.5,
+            fontSize,
+            color: textColor
+        };
+
+        return (
+            <View style={{margin: 8}}>
```

Comment:
Do we need two `View`s here or could the margin just be included in `styles.searchBar` or `searchBarStyle`?